packages/@atjson/source-html/test/converter-test.ts

```diff
@@ -541,6 +512,140 @@ describe("@atjson/source-html", () => {
       });
     });
 
+    describe("video embeds", () => {
+      describe("YouTube", () => {
+        test.each([
+          ["https://www.youtube.com/embed/0-jus6AGHzQ"],
+          ["https://www.youtube-nocookie.com/embed/0-jus6AGHzQ?controls=0"]
+        ])("%s", url => {
+          let doc = HTMLSource.fromRaw(
+            `<iframe width="560" height="315"
+            src="${url}"
+            frameborder="0"
+            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+            allowfullscreen></iframe>`
+          ).convertTo(OffsetSource);
+
+          let hir = new HIR(doc).toJSON();
+          expect(hir).toMatchObject({
+            type: "root",
+            children: [
+              {
+                type: "video-embed",
+                attributes: {
+                  url,
+                  width: 560,
+                  height: 315,
+                  aspectRatio: "16:9"
+                },
+                children: []
+              }
+            ]
+          });
+        });
+      });
+
+      describe("Vimeo", () => {
+        test("default embed code (with caption)", () => {
+          let doc = HTMLSource.fromRaw(
+            `<iframe src="https://player.vimeo.com/video/156254412" width="640" height="480" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>
+<p><a href="https://vimeo.com/156254412">TSVETOK - Vogue Italia</a> from <a href="https://vimeo.com/karimandreotti">Karim Andreotti</a> on <a href="https://vimeo.com">Vimeo</a>.</p>`
+          )
+            .convertTo(OffsetSource)
+            .canonical();
+
+          expect(doc.annotations.length).toBe(1);
+          expect(doc.annotations[0]).toMatchObject({
+            type: "video-embed",
+            attributes: {
+              url: "https://player.vimeo.com/video/156254412",
+              width: 640,
+              height: 480,
+              aspectRatio: "4:3"
+            }
+          });
+
+          let caption = doc.annotations[0].attributes.caption.canonical();
+          expect(caption.content).toBe(
+            "TSVETOK - Vogue Italia from Karim Andreotti on Vimeo."
+          );
+        });
+
+        test("embed code without caption", () => {
+          let doc = HTMLSource.fromRaw(
+            `<iframe src="https://player.vimeo.com/video/156254412" width="640" height="480" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>`
```

Comment:
I think this test (or another one) should cover the case when this iframe is followed by a paragraph that is not a Vimeo caption.

Reply:
Good call!